src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -209,13 +213,12 @@ private static bool IncludeDocument(Document document, string? clientName)
             ClientCapabilities clientCapabilities,
             CancellationToken cancellationToken)
         {
-            // Being asked about this document for the first time.  Or being asked again and we have different
-            // diagnostics.  Compute and report the current diagnostics info for this document.
-
-            // Razor has a separate option for determining if they should be in push or pull mode.
-            var diagnosticMode = document.IsRazorDocument()
-                ? InternalDiagnosticsOptions.RazorDiagnosticMode
-                : InternalDiagnosticsOptions.NormalDiagnosticMode;
+            var diagnosticMode = _serverKind switch
+            {
+                WellKnownLspServerKinds.LiveShareLspServer => InternalDiagnosticsOptions.LiveShareDiagnosticMode,
```

Comment:
note: i'm a bit confused how these options can even work now... say the option changes to push... well, you removed push support in this PR... so how could it function.

Reply:
discussed offline - this is moved to a private option in this class with no storage location.  Removing the options entirely will come when we switch roslyn entirely over to lsp pull.